File: VS/DataCommandCenter/DataCommandCenter.WebUI/src/app/models/LineageDTO.ts
import { PropertyDTO } from "./PropertyDTO";
import { Property } from "./MetadataDTOs"

export interface LineageNode {
  id: number;
  title: string;
  schemaName: string | null;
  objectName: string | null;
  objectType: string | null;
  rows: number | null;
  sizeMb: number | null;
  objectDefinition: string | null;
  description: string | null;
  metadataDictionary: Property[] | null;
  level: number | null;
}

export interface IntegrationDTO {
  id: number;
  name: string | null;
  integrationType: string | null;
  path: string | null;
  description: string | null;
  created: string | null;
  lastModified: string | null;
  metadataDictionary: Property[] | null;
}
export interface LineageLink {
  id: number;
  sourceObjectId: number | null;
  destinationObjectId: number | null;
  integrationFlowId: number | null;
  integration: IntegrationDTO;
  integrationInfo: string;
  operation: string | null;
}

export interface LineageDTO {
  nodes: LineageNode[];
  flows: LineageLink[];
}

File: VS/DataCommandCenter/DataCommandCenter.DAL/DTO/LineageDTO.cs
using DataCommandCenter.DAL.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataCommandCenter.DAL.DTO
{
    public class LineageNode
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? SchemaName { get; set; }
        public string? ObjectName { get; set; }
        public string? ObjectType { get; set; }
        public int? Rows { get; set; }
        public double? SizeMb { get; set; }
        public string? ObjectDefinition { get; set; }
        public string? Description { get; set; }
        public IEnumerable<PropertyDTO> Properties { get; set; }
        public int? Level { get; set; }
    }

    public class LineageLink
    {
        public long Id { get; set; }
        public int? SourceObjectId { get; set; }
        public int? DestinationObjectId { get; set; }
        public int? IntegrationFlowId { get; set; }

        public string IntegrationInfo { get; set; }
        public string? Operation { get; set; }
    }
    public class LineageDTO
    {
        public IEnumerable<LineageNode> Nodes { get; set; }
        public IEnumerable<LineageLink> Flows { get; set; }
    }
}
